ckan/views/user.py

```diff
@@ -497,58 +497,79 @@ def _prepare(self):
             u'user': g.user,
             u'auth_user_obj': g.userobj
         }
-        data_dict = {u'id': request.form.get(u'user')}
         try:
             logic.check_access(u'request_reset', context)
         except logic.NotAuthorized:
             base.abort(403, _(u'Unauthorized to request reset password.'))
-        return context, data_dict
 
     def post(self):
-        context, data_dict = self._prepare()
-        id = data_dict[u'id']
-
-        context = {u'model': model, u'user': g.user}
-        user_obj = None
-        try:
-            logic.get_action(u'user_show')(context, data_dict)
-            user_obj = context[u'user_obj']
-        except logic.NotFound:
-            # Try searching the user
-            if id and len(id) > 2:
-                user_list = logic.get_action(u'user_list')(context, {
-                    u'id': id
-                })
-                if len(user_list) == 1:
-                    # This is ugly, but we need the user object for the
-                    # mailer,
+        self._prepare()
+        id = request.form.get(u'user')
+        if id in (None, u''):
+            base.abort(400, _(u'"user" parameter is required'))
+        log.info(u'Password reset requested for user "{}"'.format(id))
+
+        context = {u'model': model, u'user': g.user, u'ignore_auth': True}
```

Comment:
This may seem like an obvious question, but are you adding the `u` prefix to this so it's easier for us to port to Python 3 in the near future?

Reply:
yup! :)